src/Compilers/Core/Portable/Syntax/SyntaxNodeOrToken.cs

```diff
@@ -174,14 +189,29 @@ public SyntaxNode AsNode()
             return _nodeOrParent;
         }
 
+        internal bool AsNode([NotNullWhen(true)] out SyntaxNode? node)
+        {
+            if (IsNode)
+            {
+                node = _nodeOrParent;
+                return node is object;
```

Comment:
>return node is object; [](start = 16, length = 22)

It looks like the callers of `AsNode(out SyntaxNode)` where previously using the result of `AsNode()` if `IsNode` returned `true`. Now those callers are skipping the case where `IsNode == true` but `_nodeOrParent == null`.

Is that behavior change observable and expected?

Reply:
That's true. This means really `SyntaxNodeOrToken` really has four states:
- It's a node
- It's a token
- It's default
- It's classified as a node without actually having a node

This type is plain weird. I'm not really sure how to reconcile this.